File: examples/key/new_managed_key/new_managed_key.java
KeyClient keyClient = new KeyClient();
KeyProtectionLevel keyProtectionLevel = KeyProtectionLevel.SOFTWARE;

// Generate a EcP256k key
KeyType keyType = KeyType.EcP256k;
ManagedKey managedKey = keyClient.newManagedKey(
    new ManagedKeyParams(keyProtectionLevel, keyType)
);

// Generate a Rsa2048 key
KeyType keyType = KeyType.Rsa2048;
ManagedKey managedKey = keyClient.newManagedKey(
    new ManagedKeyParams(keyProtectionLevel, keyType)
);

// Generate a Rsa3072 key
KeyType keyType = KeyType.Rsa3072;
ManagedKey managedKey = keyClient.newManagedKey(
    new ManagedKeyParams(keyProtectionLevel, keyType)
);

// Generate a Rsa4096 key
KeyType keyType = KeyType.Rsa4096;
ManagedKey managedKey = keyClient.newManagedKey(
    new ManagedKeyParams(keyProtectionLevel, keyType)
);

// Generate a Aes128 key
KeyType keyType = KeyType.Aes128;
ManagedKey managedKey = keyClient.newManagedKey(
    new ManagedKeyParams(keyProtectionLevel, keyType)
);

// Generate a Aes256 key
KeyType keyType = KeyType.Aes256;
ManagedKey managedKey = keyClient.newManagedKey(
    new ManagedKeyParams(keyProtectionLevel, keyType)
);

File: examples/key/new_managed_key/new_managed_key.py
from bloock.entity.key.key_type import KeyType
from bloock.client.key import KeyClient
from bloock.entity.key.key_protection_level import KeyProtectionLevel
from bloock.entity.key.managed_key_params import ManagedKeyParams

if __name__ == "__main__":
    key_client = KeyClient()
    protection = KeyProtectionLevel.SOFTWARE

    # Generate a EcP256k key
    key_type = KeyType.EcP256k
    managed_key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

    # Generate a Rsa2048 key
    key_type = KeyType.Rsa2048
    managed_key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

    # Generate a Rsa3072 key
    key_type = KeyType.Rsa3072
    managed_key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

    # Generate a Rsa4096 key
    key_type = KeyType.Rsa4096
    managed_key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

    # Generate a Aes128 key
    key_type = KeyType.Aes128
    managed_key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

    # Generate a Aes256 key
    key_type = KeyType.Aes256
    managed_key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

    # Generate a Bjj key
    key_type = KeyType.Bjj
    managed_key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

